unstructured_ingest/v2/processes/connectors/sql/snowflake.py

```diff
@@ -132,18 +133,22 @@ class SnowflakeUploader(PostgresUploader):
 
     def upload_contents(self, path: Path) -> None:
         df = pd.read_json(path, orient="records", lines=True)
-        df.replace({np.nan: None}, inplace=True)
 
         columns = list(df.columns)
-        stmt = f"INSERT INTO {self.upload_config.table_name} ({','.join(columns)}) VALUES({','.join([self.values_delimiter for x in columns])})"  # noqa E501
-
+        stmt = "INSERT INTO {table_name} ({columns}) VALUES({values})".format(
+            table_name=self.upload_config.table_name,
+            columns=",".join(columns),
+            values=",".join([self.values_delimiter for _ in columns]),
+        )
         for rows in pd.read_json(
             path, orient="records", lines=True, chunksize=self.upload_config.batch_size
         ):
+            rows.replace({np.nan: None}, inplace=True)
             with self.connection_config.get_cursor() as cursor:
                 values = self.prepare_data(columns, tuple(rows.itertuples(index=False, name=None)))
                 # TODO: executemany break on 'Binding data in type (list) is not supported'
                 for val in values:
+                    logger.debug(f"running query: {stmt}\nwith values: {val}")
                     cursor.execute(stmt, val)
```

Comment:
@potter-potter @rbiseck3 
We load the whole file at the beginning to read columns names, but then we load it again in batches. I feel like we could optimize it and just read columns from batches.

Reply:
The logic was actually fixed in the root sql implementation but I forgot to update this one. Take a look at this and reproduce it in the snowflake file as well: [sql.py#L377](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/sql/sql.py#L377)